sensing/autoware_pointcloud_preprocessor/src/downsample_filter/pickup_based_voxel_grid_downsample_filter_node.cpp

```diff
@@ -137,6 +136,32 @@ void PickupBasedVoxelGridDownsampleFilterComponent::filter(
   output.is_dense = input->is_dense;
   output.width = static_cast<uint32_t>(output.data.size() / output.height / output.point_step);
   output.row_step = static_cast<uint32_t>(output.data.size() / output.height);
+}
+
+void downsample_with_voxel_grid(
+  const sensor_msgs::msg::PointCloud2::ConstSharedPtr & input, const VoxelSize & voxel_size,
```

Comment:
This function does not have anything to do with ownership, so a `const&` to the pointcloud is more idiomatic.
```suggestion
  const sensor_msgs::msg::PointCloud2 & input, const VoxelSize & voxel_size,
```